manifests/logical_volume.pp

```diff
@@ -1,35 +1,85 @@
+# @param stripes The number of stripes to allocate for the new logical volume.
+#
+# @param stripesize  The stripesize to use for the new logical volume.
+#
+# @param readahead The readahead count to use for the new logical volume.
+#
+# @param range Default value: false - Set to true if the ‘size’ parameter specified, is just the minimum size you need 
+# (if the LV found is larger then the size requests this is just logged not causing a FAIL)
+#
+# @param size_is_minsize Lists strings for access control for connection method, users, databases, IPv4 addresses;
+# @param type Configures the logical volume type. AIX only
+#
+# @param thinpool Default value: false - Set to true to create a thin pool or to pool name to create thin volume
+# @param poolmetadatasize Set the initial size of the logical volume pool metadata on creation
+
+# @param mirror The number of mirrors of the volume.
+# @param mirrorlog How to store the mirror log (Allowed values: core, disk, mirrored). 
+
+# @param no_sync An optimization in lvcreate, at least on Linux.
+
+# @param region_size A mirror is divided into regions of this size (in MB), the mirror log uses this granularity to track which regions 
+# are in sync. 
+# CAN NOT BE CHANGED on already mirrored volume. 
+# Take your mirror size in terabytes and round up that number to the next power of 2, using that number as the -R argument
+
+# @param alloc Selects the allocation policy when a command needs to allocate Physical Extents from the Volume Group. Allowed Values:
+# :anywhere
+# :contiguous
+# :cling
+# :inherit
+# :normal
+
 #
 define lvm::logical_volume (
-  $volume_group,
-  $size                              = undef,
-  $initial_size                      = undef,
-  Enum['absent', 'present'] $ensure  = present,
-  $options                           = 'defaults',
-  $pass                              = '2',
-  $dump                              = '0',
-  $fs_type                           = 'ext4',
-  $mkfs_options                      = undef,
-  Stdlib::Absolutepath $mountpath    = "/${name}",
-  Boolean $mountpath_require         = false,
-  Boolean $mounted                   = true,
-  Boolean $createfs                  = true,
-  $extents                           = undef,
-  $stripes                           = undef,
-  $stripesize                        = undef,
-  $readahead                         = undef,
-  $range                             = undef,
-  $size_is_minsize                   = undef,
-  $type                              = undef,
-  Variant[Boolean, String] $thinpool = false,
-  $poolmetadatasize                  = undef,
-  $mirror                            = undef,
-  $mirrorlog                         = undef,
-  $no_sync                           = undef,
-  $region_size                       = undef,
-  $alloc                             = undef,
+  String[1] $volume_group,
+  Optional[String[1]] $size                                                     = undef,
```

Comment:
I didn't dig through the whole code, but maybe it makes sense to allow Integer for size

Reply:
`size` is the size of the logical volume. It is passed in the format `20M`, `5G`, `15G`, (you can see some more details [here](https://forge.puppet.com/modules/puppetlabs/lvm/readme) in the readme)